diff --git a/node_modules/react-native-worklets/android/.gradle/8.9/checksums/checksums.lock b/node_modules/react-native-worklets/android/.gradle/8.9/checksums/checksums.lock
new file mode 100644
index 0000000..376e931
Binary files /dev/null and b/node_modules/react-native-worklets/android/.gradle/8.9/checksums/checksums.lock differ
diff --git a/node_modules/react-native-worklets/android/.gradle/8.9/dependencies-accessors/gc.properties b/node_modules/react-native-worklets/android/.gradle/8.9/dependencies-accessors/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-worklets/android/.gradle/8.9/fileChanges/last-build.bin b/node_modules/react-native-worklets/android/.gradle/8.9/fileChanges/last-build.bin
new file mode 100644
index 0000000..f76dd23
Binary files /dev/null and b/node_modules/react-native-worklets/android/.gradle/8.9/fileChanges/last-build.bin differ
diff --git a/node_modules/react-native-worklets/android/.gradle/8.9/fileHashes/fileHashes.lock b/node_modules/react-native-worklets/android/.gradle/8.9/fileHashes/fileHashes.lock
new file mode 100644
index 0000000..6e60321
Binary files /dev/null and b/node_modules/react-native-worklets/android/.gradle/8.9/fileHashes/fileHashes.lock differ
diff --git a/node_modules/react-native-worklets/android/.gradle/8.9/gc.properties b/node_modules/react-native-worklets/android/.gradle/8.9/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-worklets/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock b/node_modules/react-native-worklets/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock
new file mode 100644
index 0000000..698bb85
Binary files /dev/null and b/node_modules/react-native-worklets/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock differ
diff --git a/node_modules/react-native-worklets/android/.gradle/buildOutputCleanup/cache.properties b/node_modules/react-native-worklets/android/.gradle/buildOutputCleanup/cache.properties
new file mode 100644
index 0000000..9b5da72
--- /dev/null
+++ b/node_modules/react-native-worklets/android/.gradle/buildOutputCleanup/cache.properties
@@ -0,0 +1,2 @@
+#Mon Jun 29 11:53:44 IST 2026
+gradle.version=8.9
diff --git a/node_modules/react-native-worklets/android/.gradle/vcs-1/gc.properties b/node_modules/react-native-worklets/android/.gradle/vcs-1/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-worklets/android/.project b/node_modules/react-native-worklets/android/.project
new file mode 100644
index 0000000..77249e6
--- /dev/null
+++ b/node_modules/react-native-worklets/android/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-worklets</name>
+	<comment>Project react-native-worklets created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1781875919618</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/react-native-worklets/android/.settings/org.eclipse.buildship.core.prefs b/node_modules/react-native-worklets/android/.settings/org.eclipse.buildship.core.prefs
new file mode 100644
index 0000000..f9ead82
--- /dev/null
+++ b/node_modules/react-native-worklets/android/.settings/org.eclipse.buildship.core.prefs
@@ -0,0 +1,13 @@
+arguments=--init-script /var/folders/vx/lmyvjcr91xq4bxhpnfrnn9vh0000gn/T/db3b08fc4a9ef609cb16b96b200fa13e563f396e9bb1ed0905fdab7bc3bc513b.gradle --init-script /var/folders/vx/lmyvjcr91xq4bxhpnfrnn9vh0000gn/T/52cde0cfcf3e28b8b7510e992210d9614505e0911af0c190bd590d7158574963.gradle --init-script /var/folders/vx/lmyvjcr91xq4bxhpnfrnn9vh0000gn/T/861a75667e10803d304a058d833cb7404195ca44013d0d61d3b653eb084379b8.gradle --init-script /var/folders/vx/lmyvjcr91xq4bxhpnfrnn9vh0000gn/T/68eb1b6516fe21c6fbba58e63c99c3207ccfc918360613709367eecde56fa77f.gradle --init-script /var/folders/vx/lmyvjcr91xq4bxhpnfrnn9vh0000gn/T/da64152279c70a8b4f3de4ca9ea66fd3b3405b7aca4e1f20f2d08e5593aa1ce1.gradle
+auto.sync=false
+build.scans.enabled=false
+connection.gradle.distribution=GRADLE_DISTRIBUTION(VERSION(8.9))
+connection.project.dir=
+eclipse.preferences.version=1
+gradle.user.home=
+java.home=/opt/homebrew/Cellar/openjdk/26.0.1/libexec/openjdk.jdk/Contents/Home
+jvm.arguments=
+offline.mode=false
+override.workspace.settings=true
+show.console.view=true
+show.executions.view=true
diff --git a/node_modules/react-native-worklets/android/build.gradle b/node_modules/react-native-worklets/android/build.gradle
index e189e1c..1ab10f0 100644
--- a/node_modules/react-native-worklets/android/build.gradle
+++ b/node_modules/react-native-worklets/android/build.gradle
@@ -34,7 +34,7 @@ def resolveReactNativeDirectory() {
     // Fallback to node resolver for custom directory structures like monorepos.
     def reactNativePackage = file(providers.exec {
         workingDir(rootDir)
-        commandLine("node", "--print", "require.resolve('react-native/package.json')")
+        commandLine(project.findProperty("NODE_BINARY") ?: "node", "--print", "require.resolve('react-native/package.json')")
     }.standardOutput.asText.get().trim())
     if (reactNativePackage.exists()) {
         return reactNativePackage.parentFile
@@ -319,7 +319,7 @@ android {
 
 def validateReactNativeVersionResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
+    commandLine(project.findProperty("NODE_BINARY") ?: "node", "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
     ignoreExitValue = true
 }
 
