diff --git a/node_modules/react-native-css-interop/src/runtime/components.ts b/node_modules/react-native-css-interop/src/runtime/components.ts
index aaf822d..e6513e9 100644
--- a/node_modules/react-native-css-interop/src/runtime/components.ts
+++ b/node_modules/react-native-css-interop/src/runtime/components.ts
@@ -67,7 +67,4 @@ remapProps(VirtualizedList, {
 
 try {
   const SafeAreaView = require("react-native-safe-area-context").SafeAreaView;
-  cssInterop(SafeAreaView, {
-    className: "style",
-  });
 } catch {}
diff --git a/node_modules/react-native-css-interop/src/runtime/native/appearance-observables.ts b/node_modules/react-native-css-interop/src/runtime/native/appearance-observables.ts
index e0fb8a0..51eda0c 100644
--- a/node_modules/react-native-css-interop/src/runtime/native/appearance-observables.ts
+++ b/node_modules/react-native-css-interop/src/runtime/native/appearance-observables.ts
@@ -26,7 +26,11 @@ export const colorScheme = {
       if ((Platform.constants?.reactNativeVersion?.minor ?? 0) >= 82) {
         appearance.setColorScheme("unspecified" as any);
       } else {
-        appearance.setColorScheme(null);
+        if ((Platform.constants?.reactNativeVersion?.minor ?? 0) >= 82) {
+          appearance.setColorScheme("unspecified" as any);
+        } else {
+          appearance.setColorScheme(null);
+        }
       }
     } else {
       appearance.setColorScheme(value);
